libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
```

Comment:
Same here - IMHO the `optional` is just useless complexity, a pointer should be enough (and simpler):

```suggestion
    const BI_NetLine* nextNetline = nullptr;
```

Reply:
I'm allergic to raw pointers :wink: :crab: 

I'll change it.